api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_online.py

```diff
@@ -40,22 +41,38 @@ def update_single_batch(self, var_vec: npt.NDArray[np.int64], val_vec: npt.NDArr
     def finalize(
         self,
     ) -> Tuple[npt.NDArray[np.float64], npt.NDArray[np.float64], npt.NDArray[np.float64], npt.NDArray[np.float64]]:
-        # TODO: do we want to add ddof param?  Currently hard-wired to 1 (see var calcs)
-
         # correct each batch to account for sparsity
         _mbomv_sparse_correct_batches(self.n_batches, self.n_samples, self.n, self.u, self.M2)
 
         # compute u, var for each batch
         batches_u = self.u
-        batches_var = (self.M2.T / (self.n_samples - 1)).T
+
+        # Note: if N-ddof is less than or equal to 0, we will return Inf - this is consistent
+        # with the numpy.var behavior.
+        batches_var = (self.M2.T / np.maximum(np.zeros(self.n_samples.size), (self.n_samples - self.ddof))).T
```

Comment:
why does this allocate an entire array of zeros for the `minimum` operation? Waste of memory.

Just use normal ufunc semantics and do something like:
> np.maximum(0,  (self.n_samples - self.ddof))).T